cfme/tests/infrastructure/test_vm_retirement_rest.py

```diff
@@ -22,10 +22,48 @@ def vm(request, provider, appliance):
     return _vm(request, provider, appliance)
 
 
+@pytest.fixture
+def retire_vm(appliance, vm, provider):
+    retire_vm = appliance.collections.infra_vms.instantiate(vm, provider)
+    # retiring VM via UI, because retiring it via API will not generate request
+    # and we will not get the retirement requester.
+    retire_vm.retire()
+
+    # using rest entity to check if the VM has retired since it is a lot faster
+    _retire_vm = appliance.rest_api.collections.vms.get(name=vm)
+    wait_for(
+        lambda: (hasattr(_retire_vm, "retired") and _retire_vm.retired),
+        timeout=1000,
+        delay=5,
+        fail_func=_retire_vm.reload,
+    )
+    return vm
+
+
+@pytest.fixture
+def vm_retirement_report(appliance, retire_vm):
+    # Create a report for Virtual Machines that exactly matches with
+    # the name of the vm that was just retired
+    report_data = {
+        "menu_name": "vm_retirement_requester",
+        "title": "VM Retirement Requester",
+        "base_report_on": "Virtual Machines",
+        "report_fields": ["Name", "Retirement Requester", "Retirement State"],
+        "filter": {
+            "primary_filter": "fill_field(Virtual Machine : Name, =, {})".format(
+                retire_vm
+            )
+        },
+    }
+    report = appliance.collections.reports.create(**report_data)
+    yield retire_vm, report
+    report.delete()
+
```

Comment:
Please check if ```delete_if_exists``` method is available for ```report```.

Reply:
Yes there is, but I am not performing any destructive activity with the report, there is no possibility that the report will be deleted during the test. Why do you it's better to use `delete_if_exists` rather than `delete` method?

Reply:
Optional:
I would use `delete_if_exists` as part of `tear-down`.  may be this fixture going to used by other test where someone do destructive activity. what you say?